hamilton/graph.py

```diff
@@ -95,43 +95,51 @@ def create_function_graph(
 
 def create_graphviz_graph(
     nodes: Set[node.Node],
-    user_nodes: Set[node.Node],
     comment: str,
     graphviz_kwargs: dict,
     node_modifiers: Dict[str, dict],
+    strictly_display_only_nodes_passed_in: bool,
 ) -> "graphviz.Digraph":  # noqa: F821
     """Helper function to create a graphviz graph.
 
     :param nodes: The set of computational nodes
-    :param user_nodes: The set of nodes that the user is providing inputs for.
     :param comment: The comment to have on the graph.
     :param graphviz_kwargs: kwargs to pass to create the graph.
         e.g. dict(graph_attr={'ratio': '1'}) will set the aspect ratio to be equal of the produced image.
     :param node_modifiers: A dictionary of node names to dictionaries of node attributes to modify.
+    :param strictly_display_only_nodes_passed_in: If True, only display the nodes passed in. Else defaults to displaying
+        also what nodes a node depends on (i.e. all nodes that feed into it).
     :return: a graphviz.Digraph; use this to render/save a graph representation.
     """
     import graphviz
 
     digraph = graphviz.Digraph(comment=comment, **graphviz_kwargs)
     for n in nodes:
+        label = n.name
         other_args = {}
         # checks if the node has any modifiers
         if n.name in node_modifiers:
             # if node is an output, then modify the node to be a rectangle
             if node_modifiers[n.name].get("is_output"):
                 other_args["shape"] = "rectangle"
-        digraph.node(n.name, label=n.name, **other_args)
-    for n in user_nodes:
-        other_args = {"style": "dashed"}
-        # checks if the node has any modifiers
-        if n.name in node_modifiers:
-            # if node is an output, then modify the node to be a rectangle
-            if node_modifiers[n.name].get("is_output"):
-                other_args["shape"] = "rectangle"
-        digraph.node(n.name, label=f"Input: {n.name}", **other_args)
-
-    for n in list(nodes) + list(user_nodes):
+            if node_modifiers[n.name].get("is_path"):
+                other_args["color"] = "red"
```

Comment:
Make a constant maybe?

Reply:
it's not used anywhere else - if it was -- then yes.